app/models/concerns/marc_links.rb

```diff
@@ -52,6 +52,20 @@ def finding_aid?
       link_struct[:material_type]&.downcase&.include?('finding aid') || link_struct[:note]&.downcase&.include?('finding aid')
     end
 
+    def archives_finding_aid?
+      finding_aid? && link_struct[:href]&.include?('//archives.stanford.edu')
+    end
+
+    def finding_aid_link_text
+      archives_finding_aid? ? 'Archival Collections at Stanford' : 'Online Archive of California'
+    end
```

Comment:
If we're now distinguishing between finding aid hosts, maybe it's time to extract something like `oac_finding_aid?` too?

```suggestion
    def archives_finding_aid?
      finding_aid? && link_struct[:href]&.include?('//archives.stanford.edu')
    end

    def oac_finding_aid?
      finding_aid? && link_struct[:href]&.include?('oac.cdlib.org')
    end

    def finding_aid_link_text
      if archives_finding_aid?
        'Archival Collections at Stanford'
      elsif oac_finding_aid?
        'Online Archive of California'
      else
        link_struct[:link_text] || link_host
      end   
    end
```